app/src/main/java/com/dscnsut/app/blogs/BListAdapter.kt

```diff
@@ -0,0 +1,64 @@
+package com.dscnsut.app.blogs
+
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.ViewModel
+import androidx.navigation.findNavController
+import androidx.recyclerview.widget.RecyclerView
+import com.dscnsut.app.R
+import java.util.*
+
+class BListAdapter(private val listener : OnItemClickListener): RecyclerView.Adapter<BListAdapter.BListVHolder>()  {
+    var mutableBlogList = mutableListOf<BlogsModel>()
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BListVHolder {
+
+        val itemView = LayoutInflater.from(parent.context).inflate(R.layout.bloglist_layout,parent,false)
+
+        return BListVHolder(itemView)
+    }
+
+    override fun onBindViewHolder(holder: BListVHolder, position: Int) {
+        val ci = mutableBlogList[position]
+
+        holder.tv1.text = ci.title
+        holder.tv2.text = ci.timestamp.toString()
+        holder.tv3.text = ci.author
+
```

Comment:
make a cardView object in the view holder class.
in that object set the onClickListener like this:
```kotlin
holder.cardView.setOnClickListener() {
    listener.onItemClick(ci)
}
```